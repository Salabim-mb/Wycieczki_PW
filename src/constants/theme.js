const theme = {
    color: {
        primary: '#333',
        secondary: '#888',
        green: '#6ABA9C',
        darkBlue: '#3C3C4C',
        yellow: '#FED542',
        orange: '#EB7C5A',
        blue: '#7796CD',
        darkPurple: '#966078',
        purple: '#B4A0AA'
    },
    fontSize: {
        s: '0.75rem',
        xs: '1rem',
        m: '1.25rem',
        xm: '1.5rem',
        l: '1.75rem',
        xl: '2rem',
    },
    margin: {
        s: '0.5rem',
        xs: '0.75rem',
        m: '1rem',
        xm: '1.25rem',
        l: '1.5rem',
        xl: '1.75rem',
    },
    zIndex: {
        max: 3,
        medium: 2,
    },
    breakpoint: {
        s: 460,
        m: 768,
        l: 1280,
        xl: 1920,
    },
    time: {
        shortest: 150,
        shorter: 200,
        short: 250,
        standard: 300,
        long: 350,
        longer: 400,
        longest: 450,
    }
}

export default theme