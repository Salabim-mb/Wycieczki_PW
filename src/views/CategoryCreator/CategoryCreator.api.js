import paths from 'constants/api';


export const getBlogs = async () => {
    const url = `${paths.BLOGS}`;

    const headers = {
        'Content-Type': 'application/json',
    };

    const response = await fetch(url, { headers, method: 'GET' });

    if (response.status !== 200) {
        throw new Error('błąd');
    }

    return response.json();
}


export const getCategory = async (id) => {
    if (!id) return null

    const url = `${paths.BLOG_TOPIC}${id}`;

    const headers = {
        'Content-Type': 'application/json',
    };

    const response = await fetch(url, { headers, method: 'GET' });

    if (response.status !== 200) {
        throw new Error('błąd');
    }

    return response.json();
}

// export const sendCategory = async (formData) => {
//     const url = `${paths.BLOG_TOPIC}`;

//     const headers = {
//         'Content-Type': 'multipart/form-data',
//     };

//     const response = await fetch(url, { headers, method: 'POST', body: formData });

//     console.log(response)

//     if (response.status !== 200) {
//         throw new Error('błąd');
//     }

//     return response.json();
// }

export const sendCategory = (formData, id) => {
    let methodType = 'POST'
    let url = `${paths.BLOG_TOPIC}`

    if (id) {
        methodType = 'PATCH'
        url = `${paths.BLOG_TOPIC}${id}/`
    }

    console.log(methodType, id)

    fetch(url, {
        method: methodType,

        body: formData,
    })
        .then((response) => {
            if (response.ok) {
                return response.json();
            }
            throw new Error('Something went wrong');

        })
        // .then(res => console.log(res))
        .catch(err => err.message)
}