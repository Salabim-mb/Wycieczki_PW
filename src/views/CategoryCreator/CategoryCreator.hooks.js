import { useQuery, useMutation } from 'react-query';
import { getBlogs, sendCategory, getCategory } from './CategoryCreator.api'

export const useQueryBlogs = () => useQuery('blogs', getBlogs, { retry: false })
export const useQueryCategory = (id) => useQuery('categoryGet', () => getCategory(id), { retry: false },)
export const useMutationCategory = (id) => useMutation((formData) => sendCategory(formData, id), { retry: false })