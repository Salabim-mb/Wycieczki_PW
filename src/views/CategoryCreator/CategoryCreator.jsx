import React from 'react'
import styled from 'styled-components'
import { useFormik } from 'formik';
import { UploadButton, AlertInfo, Input } from 'components';
import { PhotoCamera } from '@material-ui/icons';
import Button from '@material-ui/core/Button';
import Select from '@material-ui/core/Select';
import { Alert } from '@material-ui/lab';
import CenteredContainer from 'components/CenteredContainer'
import { useQueryBlogs } from './CategoryCreator.hooks'
import { sendCategory } from './CategoryCreator.api'

const StyledForm = styled.form`
    display:flex;
    flex-direction: column;
    justify-content: center;
    background-color: white;
    padding: ${({ theme }) => theme.margin.l};

    @media (min-width: 1024px){
        width: 50%;
    }
`;

const CategoryCreator = () => {
    const { data, isError, isLoading, error } = useQueryBlogs()
    // const mutation = useMutationCategory()


    const formik = useFormik({
        initialValues: {
            title: '',
            blog: 0,
            description: '',
            cover_image: '',
            header_image: ''
        },
        onSubmit: async () => {
            const formData = new FormData();
            formData.append('cover_image', formik.values.cover_image);
            formData.append('header_image', formik.values.header_image);
            formData.append('title', formik.values.title);
            formData.append('blog', formik.values.blog);
            formData.append('description', formik.values.description);
            // formData.append('authorization', user.token);
            sendCategory(formData)
            // try {
            //     const todo = await mutation.mutateAsync(formData)
            //     console.log(todo)
            // } catch (err) {
            //     console.error(err)
            // } finally {
            //     console.log('done')
            // }
        },
        validate: (values) => {
            const errors = {};

            if (!values.title) {
                errors.title = 'Pole wymagane';
            } else if (values.title.length > 100) {
                errors.title = 'Za długie';
            }
            if (!values.blog) {
                errors.blog = 'Pole wymagane';
            }

            return errors;
        }
    });


    return (
        <CenteredContainer>
            <h2>Dodaj kategorię</h2>
            <StyledForm onSubmit={formik.handleSubmit}>
                <Input
                    fullWidth
                    id="title"
                    name="title"
                    label="Tytuł"
                    variant="outlined"
                    value={formik.values.title}
                    inputProps={{ minLength: 1, maxLength: 100, 'data-testid': 'test-id' }}
                    handleChange={formik.handleChange}
                />
                {formik.touched.title && formik.errors.title && (<Alert severity="error">Nie może być puste i dłuższe niż 100 znaków</Alert>)}
                <Input
                    fullWidth
                    id="description"
                    name="description"
                    label="Opis"
                    variant="outlined"
                    value={formik.values.description}
                    handleChange={formik.handleChange}
                />
                <Select
                    id="blog"
                    native
                    value={formik.values.blog}
                    onChange={formik.handleChange}
                >
                    <option value="" selected>-----------</option>
                    {isError ? <option value="">Nie udało się pobrać blogów</option> : data?.map((blog) => (
                        <option key={blog.id} value={blog.id}>{blog.title}</option>
                    ))}
                </Select>
                <AlertInfo isLoading={isLoading} isError={isError}>
                    {error?.message}
                </AlertInfo>

                {formik.touched.blog && formik.errors.blog && (<Alert severity="error">Nie może być puste.</Alert>)}

                <UploadButton
                    header="Prześlij cover"
                    accept="image/*"
                    icon={<PhotoCamera />}
                    id="cover_image"
                    name="cover_image"
                    handleChange={event => {
                        formik.setFieldValue('cover_image', event.currentTarget.files[0]);
                    }}
                    error={formik.touched.cover && formik.errors.cover ? formik.errors.cover : ''}
                />
                {/* <Input id="cover_image"
                    name="cover_image"
                    type="file" onChange={event => {
                        formik.setFieldValue('cover_image', event.currentTarget.files[0]);
                    }} /> */}

                <UploadButton
                    header="Prześlij cover"
                    accept="image/*"
                    icon={<PhotoCamera />}
                    id="header_image"
                    name="header_image"
                    handleChange={event => {
                        formik.setFieldValue('header_image', event.currentTarget.files[0]);
                    }}
                    error={formik.touched.cover && formik.errors.cover ? formik.errors.cover : ''}
                />

                {/* <Input id="header_image"
                    name="header_image"
                    type="file" onChange={event => {
                        formik.setFieldValue('header_image', event.currentTarget.files[0]);
                    }} /> */}
                <Button type="submit" onSubmit={formik.handleSubmit}>Zatwierdź</Button>
            </StyledForm>

        </CenteredContainer>
    )
}

export default CategoryCreator