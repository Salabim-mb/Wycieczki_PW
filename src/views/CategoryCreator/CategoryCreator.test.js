import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';
import theme from 'constants/theme';
import CategoryCreator from './CategoryCreator';
import { useQueryBlogs } from './CategoryCreator.hooks';

jest.mock('views/CategoryCreator/CategoryCreator.hooks', () => ({
    useQueryBlogs: jest.fn(),
}));

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useParams: () => ({
        id: 'abc',
    }),
}));

describe('CategoryCreator', () => {
    beforeEach(() => {
        jest.clearAllMocks();
    });

    it('should match snapshot - data', async () => {
        useQueryBlogs.mockImplementation(() => ({
            isSuccess: true,
            data: [{
                id: 1,
                index: 1,
                content: 'aaa',
                image_url: '/link',
                title: 'bbb',
                show_title: true,
                color: '#000'
            }],
            isError: false,
            isLoading: false,
        }));
        const { container } = render(
            <MemoryRouter>
                <ThemeProvider theme={theme}>
                    <CategoryCreator />
                </ThemeProvider>
            </MemoryRouter>,
        );

        expect(container).toMatchSnapshot();
    });

    it('should match snapshot - isError', async () => {
        useQueryBlogs.mockImplementation(() => ({
            isSuccess: true,
            data: [],
            error: 'błąd',
            isError: true,
            isLoading: false,
        }));
        const { container } = render(
            <MemoryRouter>
                <ThemeProvider theme={theme}>
                    <CategoryCreator />
                </ThemeProvider>
            </MemoryRouter>,
        );

        expect(container).toMatchSnapshot();
    });

    it('should match snapshot - isLoading', async () => {
        useQueryBlogs.mockImplementation(() => ({
            isSuccess: true,
            data: [],
            isError: false,
            isLoading: true,
        }));
        const { container } = render(
            <MemoryRouter>
                <ThemeProvider theme={theme}>
                    <CategoryCreator />
                </ThemeProvider>
            </MemoryRouter>,
        );

        expect(container).toMatchSnapshot();
    });
});