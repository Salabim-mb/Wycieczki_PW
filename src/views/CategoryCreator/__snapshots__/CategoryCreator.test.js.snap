// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryCreator should match snapshot - data 1`] = `
<div>
  <div
    class="MuiContainer-root sc-hKgILt havGhI MuiContainer-maxWidthLg"
  >
    <h2>
      Dodaj kategorię
    </h2>
    <form
      class="sc-eCssSg cmKfWZ"
    >
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        rest="[object Object]"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink="false"
        >
          Tytuł
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input"
            data-testid="test-id"
            maxlength="100"
            minlength="1"
            name="title"
            placeholder=""
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-3"
            >
              <span>
                Tytuł
              </span>
            </legend>
          </fieldset>
        </div>
        
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        rest="[object Object]"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink="false"
        >
          Opis
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input"
            name="description"
            placeholder=""
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-3"
            >
              <span>
                Opis
              </span>
            </legend>
          </fieldset>
        </div>
        
      </div>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline"
      >
        <select
          class="MuiSelect-root MuiSelect-select MuiInputBase-input MuiInput-input"
          id="blog"
        >
          <option
            value=""
          >
            -----------
          </option>
          <option
            value="1"
          >
            bbb
          </option>
        </select>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSelect-icon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
      <label
        aria-disabled="false"
        class="MuiButtonBase-root MuiButton-root MuiButton-contained"
        role="button"
        tabindex="0"
      >
        <span
          class="MuiButton-label"
        >
          <span
            class="MuiButton-startIcon MuiButton-iconSizeMedium"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <circle
                cx="12"
                cy="12"
                r="3.2"
              />
              <path
                d="M9 2L7.17 4H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2h-3.17L15 2H9zm3 15c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5z"
              />
            </svg>
          </span>
          Prześlij cover
          <input
            accept="image/*"
            hidden=""
            id="cover_image"
            name="cover_image"
            type="file"
          />
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </label>
      
      <label
        aria-disabled="false"
        class="MuiButtonBase-root MuiButton-root MuiButton-contained"
        role="button"
        tabindex="0"
      >
        <span
          class="MuiButton-label"
        >
          <span
            class="MuiButton-startIcon MuiButton-iconSizeMedium"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <circle
                cx="12"
                cy="12"
                r="3.2"
              />
              <path
                d="M9 2L7.17 4H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2h-3.17L15 2H9zm3 15c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5z"
              />
            </svg>
          </span>
          Prześlij cover
          <input
            accept="image/*"
            hidden=""
            id="header_image"
            name="header_image"
            type="file"
          />
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </label>
      
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text"
        tabindex="0"
        type="submit"
      >
        <span
          class="MuiButton-label"
        >
          Zatwierdź
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </form>
  </div>
</div>
`;

exports[`CategoryCreator should match snapshot - isError 1`] = `
<div>
  <div
    class="MuiContainer-root sc-hKgILt havGhI MuiContainer-maxWidthLg"
  >
    <h2>
      Dodaj kategorię
    </h2>
    <form
      class="sc-eCssSg cmKfWZ"
    >
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        rest="[object Object]"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink="false"
        >
          Tytuł
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input"
            data-testid="test-id"
            maxlength="100"
            minlength="1"
            name="title"
            placeholder=""
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-7"
            >
              <span>
                Tytuł
              </span>
            </legend>
          </fieldset>
        </div>
        
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        rest="[object Object]"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink="false"
        >
          Opis
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input"
            name="description"
            placeholder=""
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-7"
            >
              <span>
                Opis
              </span>
            </legend>
          </fieldset>
        </div>
        
      </div>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline"
      >
        <select
          class="MuiSelect-root MuiSelect-select MuiInputBase-input MuiInput-input"
          id="blog"
        >
          <option
            value=""
          >
            -----------
          </option>
          <option
            value=""
          >
            Nie udało się pobrać blogów
          </option>
        </select>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSelect-icon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
      <div
        class="MuiPaper-root MuiAlert-root MuiAlert-standardError MuiPaper-elevation0"
        role="alert"
      >
        <div
          class="MuiAlert-icon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20,12A8,8 0 0,1 12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4C12.76,4 13.5,4.11 14.2, 4.31L15.77,2.74C14.61,2.26 13.34,2 12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0, 0 22,12M7.91,10.08L6.5,11.5L11,16L21,6L19.59,4.58L11,13.17L7.91,10.08Z"
            />
          </svg>
        </div>
        <div
          class="MuiAlert-message"
        />
      </div>
      <label
        aria-disabled="false"
        class="MuiButtonBase-root MuiButton-root MuiButton-contained"
        role="button"
        tabindex="0"
      >
        <span
          class="MuiButton-label"
        >
          <span
            class="MuiButton-startIcon MuiButton-iconSizeMedium"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <circle
                cx="12"
                cy="12"
                r="3.2"
              />
              <path
                d="M9 2L7.17 4H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2h-3.17L15 2H9zm3 15c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5z"
              />
            </svg>
          </span>
          Prześlij cover
          <input
            accept="image/*"
            hidden=""
            id="cover_image"
            name="cover_image"
            type="file"
          />
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </label>
      
      <label
        aria-disabled="false"
        class="MuiButtonBase-root MuiButton-root MuiButton-contained"
        role="button"
        tabindex="0"
      >
        <span
          class="MuiButton-label"
        >
          <span
            class="MuiButton-startIcon MuiButton-iconSizeMedium"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <circle
                cx="12"
                cy="12"
                r="3.2"
              />
              <path
                d="M9 2L7.17 4H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2h-3.17L15 2H9zm3 15c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5z"
              />
            </svg>
          </span>
          Prześlij cover
          <input
            accept="image/*"
            hidden=""
            id="header_image"
            name="header_image"
            type="file"
          />
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </label>
      
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text"
        tabindex="0"
        type="submit"
      >
        <span
          class="MuiButton-label"
        >
          Zatwierdź
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </form>
  </div>
</div>
`;

exports[`CategoryCreator should match snapshot - isLoading 1`] = `
<div>
  <div
    class="MuiContainer-root sc-hKgILt havGhI MuiContainer-maxWidthLg"
  >
    <h2>
      Dodaj kategorię
    </h2>
    <form
      class="sc-eCssSg cmKfWZ"
    >
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        rest="[object Object]"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink="false"
        >
          Tytuł
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input"
            data-testid="test-id"
            maxlength="100"
            minlength="1"
            name="title"
            placeholder=""
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-9 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-11"
            >
              <span>
                Tytuł
              </span>
            </legend>
          </fieldset>
        </div>
        
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        rest="[object Object]"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink="false"
        >
          Opis
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input"
            name="description"
            placeholder=""
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-9 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-11"
            >
              <span>
                Opis
              </span>
            </legend>
          </fieldset>
        </div>
        
      </div>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline"
      >
        <select
          class="MuiSelect-root MuiSelect-select MuiInputBase-input MuiInput-input"
          id="blog"
        >
          <option
            value=""
          >
            -----------
          </option>
        </select>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSelect-icon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
      <div
        class="MuiPaper-root MuiAlert-root MuiAlert-standardInfo MuiPaper-elevation0"
        role="alert"
      >
        <div
          class="MuiAlert-icon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20,12A8,8 0 0,1 12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4C12.76,4 13.5,4.11 14.2, 4.31L15.77,2.74C14.61,2.26 13.34,2 12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0, 0 22,12M7.91,10.08L6.5,11.5L11,16L21,6L19.59,4.58L11,13.17L7.91,10.08Z"
            />
          </svg>
        </div>
        <div
          class="MuiAlert-message"
        >
          Ładowanie...
        </div>
      </div>
      <label
        aria-disabled="false"
        class="MuiButtonBase-root MuiButton-root MuiButton-contained"
        role="button"
        tabindex="0"
      >
        <span
          class="MuiButton-label"
        >
          <span
            class="MuiButton-startIcon MuiButton-iconSizeMedium"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <circle
                cx="12"
                cy="12"
                r="3.2"
              />
              <path
                d="M9 2L7.17 4H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2h-3.17L15 2H9zm3 15c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5z"
              />
            </svg>
          </span>
          Prześlij cover
          <input
            accept="image/*"
            hidden=""
            id="cover_image"
            name="cover_image"
            type="file"
          />
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </label>
      
      <label
        aria-disabled="false"
        class="MuiButtonBase-root MuiButton-root MuiButton-contained"
        role="button"
        tabindex="0"
      >
        <span
          class="MuiButton-label"
        >
          <span
            class="MuiButton-startIcon MuiButton-iconSizeMedium"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <circle
                cx="12"
                cy="12"
                r="3.2"
              />
              <path
                d="M9 2L7.17 4H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2h-3.17L15 2H9zm3 15c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5z"
              />
            </svg>
          </span>
          Prześlij cover
          <input
            accept="image/*"
            hidden=""
            id="header_image"
            name="header_image"
            type="file"
          />
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </label>
      
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text"
        tabindex="0"
        type="submit"
      >
        <span
          class="MuiButton-label"
        >
          Zatwierdź
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </form>
  </div>
</div>
`;
